### PR TITLE
Initial version of base smart contract

### PR DESCRIPTION
**Pull request by vladichhh**

### CODE REVIEW

File: tests/todomanager-tests.js [hunk 1]
```diff
@@ -0,0 +1,11 @@
+const assert = require('assert');
+const eoslime = require('eoslime').init('local');
```
Comment (bakasura980) [inline tests/todomanager-tests.js]: Don't use it like that -> instead use eoslime test

File: tests/todomanager-tests.js [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+
+const TODO_MANAGER_WASM_PATH = '../contracts/todomanager.wasm';
+const TODO_MANAGER_ABI_PATH = '../contracts/todomanager.abi';
+
+describe('TODO Manager', function () {
```
Comment (bakasura980) [inline tests/todomanager-tests.js]: eoslime test will change `describe('TODO Manager', function () {` to `describe('TODO Manager', function (eoslime) {`

File: package.json [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+    "test": "test"
+  },
+  "scripts": {
+    "test": "mocha --timeout 10000 ./tests/todomanager-tests.js"
+  },
+  "repository": {
+    "type": "git",
+    "url": "git+https://github.com/LimeChain/eoslime-shape-react.git"
+  },
+  "author": "",
```
Comment (bakasura980) [inline package.json]: Add LimeChain

File: package.json [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+  "license": "ISC",
```
Comment (bakasura980) [inline package.json]: MIT

File: package.json [hunk 5]
```diff
@@ -0,0 +1,26 @@
+{
+  "name": "eoslime-shape-react",
+  "version": "1.0.0",
+  "description": "",
+  "main": "index.js",
+  "directories": {
+    "test": "test"
+  },
+  "scripts": {
+    "test": "mocha --timeout 10000 ./tests/todomanager-tests.js"
```
Comment (bakasura980) [inline package.json]: eoslime test

File: package.json [hunk 6]
```diff
@@ -0,0 +1,26 @@
+{
+  "name": "eoslime-shape-react",
+  "version": "1.0.0",
+  "description": "",
```
Comment (bakasura980) [inline package.json]: Write a simple description of what it is about

File: contracts/todomanager.cpp [hunk 7]
```diff
@@ -0,0 +1,38 @@
+#include "todomanager.hpp"
+
+namespace todomanager
+{
+
+void todomanager::addtodo(string description)
```
Comment (bakasura980) [inline contracts/todomanager.cpp]: addtodo -> add

File: contracts/todomanager.cpp [hunk 8]
```diff
@@ -0,0 +1,38 @@
...
+{
+    todos_table.emplace(_self, [&](auto &new_todo) {
+        new_todo.id = todos_table.available_primary_key();
+        new_todo.description = description;
+        new_todo.status = e_status_option::INITIAL;
+    });
+}
+
+void todomanager::updatetodo(uint64_t index, uint8_t status)
```
Comment (bakasura980) [inline contracts/todomanager.cpp]: updatetoto -> update

File: contracts/todomanager.cpp [hunk 9]
```diff
@@ -0,0 +1,38 @@
...
+
+    check(todo_iterator != todos_table.end(), "Todo does not exist");
+    check(status > todo_iterator->id, "Wrong new status");
+
+    todos_table.modify(todo_iterator, _self, [&](auto &update_todo) {
+        update_todo.status = status;
+    });
+}
+
+void todomanager::removetodo(uint64_t index)
```
Comment (bakasura980) [inline contracts/todomanager.cpp]: removetodo -> remove

File: contracts/todomanager.cpp [hunk 10]
```diff
@@ -0,0 +1,38 @@
...
+        new_todo.status = e_status_option::INITIAL;
+    });
+}
+
+void todomanager::updatetodo(uint64_t index, uint8_t status)
+{
+    auto todo_iterator = todos_table.find(index);
+
+    check(todo_iterator != todos_table.end(), "Todo does not exist");
+    check(status > todo_iterator->id, "Wrong new status");
```
Comment (bakasura980) [inline contracts/todomanager.cpp]: A todo could be moved left and right

File: contracts/todomanager.hpp [hunk 11]
```diff
@@ -0,0 +1,51 @@
...
+{
+public:
+    using contract::contract;
+
+    todomanager(name receiver, name code, datastream<const char *> ds) : contract(receiver, code, ds),
+                                                                         todos_table(receiver, code.value)
+    {
+    }
+
+    enum e_status_option : uint8_t {
```
Comment (bakasura980) [inline contracts/todomanager.hpp]: Rename it to status (what is the idea of e_status ?)

File: contracts/todomanager.hpp [hunk 12]
```diff
@@ -0,0 +1,51 @@
...
+        INITIAL = 0,
```
Comment (bakasura980) [inline contracts/todomanager.hpp]: BACKLOG

File: contracts/todomanager.hpp [hunk 13]
```diff
@@ -0,0 +1,51 @@
...
+        TODO,
```
Comment (bakasura980) [inline contracts/todomanager.hpp]: IN_PROGRESS

File: contracts/todomanager.hpp [hunk 14]
```diff
@@ -0,0 +1,51 @@
...
+        INPROGRESS,
```
Comment (bakasura980) [inline contracts/todomanager.hpp]: REVIEW

File: contracts/todomanager.hpp [hunk 15]
```diff
@@ -0,0 +1,51 @@
...
+
+    typedef uint8_t status_option;
+
+    struct [[eosio::table]] todo
+    {
+        uint64_t id;
+        string description;
+        status_option status;
+
+        uint64_t primary_key() const { return id; }
```
Comment (bakasura980) [inline contracts/todomanager.hpp:33]: Add a secondary key to search by status as we will have a board and there we should show all todos from a status group

File: contracts/todomanager.hpp [hunk 16]
```diff
@@ -0,0 +1,51 @@
...
+    enum e_status_option : uint8_t {
+        INITIAL = 0,
+        TODO,
+        INPROGRESS,
+        DONE
+    };
+
+    typedef uint8_t status_option;
+
+    struct [[eosio::table]] todo
```
Comment (bakasura980) [inline contracts/todomanager.hpp]: What about assignee ?

File: contracts/todomanager.cpp (hunk 7 above)
Comment (bakasura980) [inline contracts/todomanager.cpp]: Assignee ?

**Review by bakasura980 — CHANGES_REQUESTED** [review]

File: package.json (hunk 3 above)
Comment (vladichhh) [reply package.json]: fixed

File: package.json (hunk 4 above)
Comment (vladichhh) [reply package.json]: fixed

File: contracts/todomanager.cpp (hunk 7 above)
Comment (vladichhh) [reply contracts/todomanager.cpp]: updated

File: contracts/todomanager.cpp (hunk 8 above)
Comment (vladichhh) [reply contracts/todomanager.cpp]: updated

File: contracts/todomanager.cpp (hunk 9 above)
Comment (vladichhh) [reply contracts/todomanager.cpp]: updated

File: contracts/todomanager.hpp (hunk 13 above)
Comment (vladichhh) [reply contracts/todomanager.hpp]: updated

File: contracts/todomanager.hpp (hunk 12 above)
Comment (vladichhh) [reply contracts/todomanager.hpp]: updated

File: contracts/todomanager.hpp (hunk 14 above)
Comment (vladichhh) [reply contracts/todomanager.hpp]: updated

File: package.json (hunk 6 above)
Comment (vladichhh) [reply package.json]: added description

File: contracts/todomanager.cpp (hunk 10 above)
Comment (vladichhh) [reply contracts/todomanager.cpp]: updated check

File: contracts/todomanager.hpp (hunk 15 above)
Comment (vladichhh) [reply contracts/todomanager.hpp:33]: added secondary key by_status()

File: contracts/todomanager.hpp (hunk 11 above)
Comment (vladichhh) [reply contracts/todomanager.hpp]: updated enum type

File: tests/todomanager-tests.js (hunk 1 above)
Comment (vladichhh) [reply tests/todomanager-tests.js]: fixed

File: tests/todomanager-tests.js (hunk 2 above)
Comment (vladichhh) [reply tests/todomanager-tests.js]: fixed

File: package.json (hunk 5 above)
Comment (vladichhh) [reply package.json]: fixed

File: tests/todomanager-tests.js [hunk 17]
```diff
@@ -0,0 +1,69 @@
...
+    this.timeout(15000);
+
+    beforeEach(async () => {
+        managerContract = await eoslime.Contract.deploy(TODO_MANAGER_WASM_PATH, TODO_MANAGER_ABI_PATH);
+    });
+
+    it('Should add a new todo', async () => {
+        await managerContract.add(DESCRIPTION);
+
+        const todos = await eoslime.Provider.select(TODOS_TABLE).from(managerContract.name).find()
```
Comment (bakasura980) [inline tests/todomanager-tests.js]: managerContract.todos.find()

**Review by bakasura980 — CHANGES_REQUESTED** [review]

File: tests/todomanager-tests.js (hunk 17 above)
Comment (vladichhh) [reply tests/todomanager-tests.js]: updated

**Pull request merged**